bpf/lib/nodeport_egress.h

```diff
@@ -306,13 +306,34 @@ static __always_inline int nodeport_snat_fwd_ipv4(struct __ctx_buff *ctx,
 	void *data, *data_end;
 	struct iphdr *ip4;
 	int l4_off, ret;
+	struct endpoint_info *ep;
 
 	if (!revalidate_data(ctx, &data, &data_end, &ip4))
 		return DROP_INVALID;
 
 	snat_v4_init_tuple(ip4, NAT_DIR_EGRESS, &tuple);
 	l4_off = ETH_HLEN + ipv4_hdrlen(ip4);
 
+	ep = __lookup_ip4_endpoint(ip4->saddr);
+	if (ep && ep->parent_ifindex && ep->parent_ifindex != NATIVE_DEV_IFINDEX) {
+		/* This packet came from an endpoint with a parent interface and
+		 * it us currently not egressing on its parent interface.
+		 * Check if its a reply packet, if it is, redirect it to the
+		 * parent interface.
+		 */
+		if (l4_load_ports(ctx, l4_off, (__be16 *)&tuple.dport) < 0)
+			return DROP_INVALID;
+
+		if (ct_is_reply4(get_ct_map4(&tuple), &tuple)) {
+			union macaddr smac = NATIVE_DEV_MAC_BY_IFINDEX(ep->parent_ifindex);
+
+			if (eth_store_saddr_aligned(ctx, smac.addr, 0) < 0)
+				return DROP_WRITE_ERROR;
+
+			return ctx_redirect(ctx, ep->parent_ifindex, 0);
```

Comment:
This assumes that the packet will still take the same next-hop when egressing via the parent interface, correct? I think that's worth pointing out here with a code comment. Although I'm not sure it's a safe assumption?

Reply:
At least in the case of EKS this is valid, I checked that all ENI's have the same gateway with the same MAC address. All interfaces essentially lives on the same L2 network from our perspective. I will add a comment explaining that as well.

Reply:
Sounds good! Once we bumped our minimal kernel requirement, we could easily switch to `bpf_redirect_neigh()`. Just to be on the safe side.

(this leaves in the egress path, so XDP is no concern)